lib/urlwatch/jobs.py

```diff
@@ -230,3 +231,29 @@ def retrieve(self, job_state):
                 return response.content.decode('ascii', 'ignore')
 
         return response.text
+
+class FileJob(Job):
+    """Read the content of a local file"""
+
+    __kind__ = 'file'
+    __required__ = ('path',)
+
+    def get_location(self):
+        return str(self.path)
+
+    def retrieve(self, job_state):
+
+        path_string = str(self.path)
+
+        if not path_string:
+            raise ValueError('Path cannot be null or empty')
+
+        if path_string.startswith('file://'):
```

Comment:
OK I got it now. But in my opinion it would be a cleaner solution implementing a transport adapter, so in the future we can further extend the supported protocols. I can prepare a proposal if you want.